lib/omniauth/strategy.rb

```diff
@@ -462,6 +466,9 @@ def query_string
 
     def call_app!(env = @env)
       @app.call(env)
+    rescue StandardError => e
```

Comment:
I was wondering if it would be possible to rescue a subclass of `StandardError` like `OmniAuthError` or something like that.  I guess the challenge is that `call_app` is called from the other dependent gems (omniauth-saml, etc) and you can't really know what will be raised.